File: backend/tour-design-api/src/main/java/com/tourdesign/platform/model/TravelPackageModel.java
package com.tourdesign.platform.model;

import lombok.*;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TravelPackageModel {
    private Long id;

    private String name;
    private String description;
    private Integer durationDays;
    private Double estimatedPrice;
    private String focusType;
    private Integer difficultyLevel;
    private String targetAudience;
    private String serviceLanguage;

    private List<PackageHotelModel> hotels;
    private List<PackageRestaurantModel> restaurants;
    private List<PackageSpotModel> spots;
}

File: frontend/tour-design-web/src/app/models/travel-package.model.ts
// src/app/models/travel-package.model.ts
import { PackageHotelModel } from './package-hotel.model';
import { PackageRestaurantModel } from './package-restaurant.model';
import { PackageSpotModel } from './package-spot.model';

export interface TravelPackageModel {
  id: number;
  name: string;
  description?: string;
  durationDays?: number;
  estimatedPrice?: number;
  focusType?: string;
  difficultyLevel?: number;
  targetAudience?: string;
  serviceLanguage?: string;

  hotels?: PackageHotelModel[];
  restaurants?: PackageRestaurantModel[];
  spots?: PackageSpotModel[];
}
